### PR TITLE
docs: add v3 OpenAPI specifications

### PR DESCRIPTION
**Pull request by jlabatut**
Update OpenAPI specifications, in order to comply with Polyflix v3 :

- Courses are the only entity now.
- Elements will be stored in `sections`.
- Elements can be `VIDEO`, `QUIZ` and `PAGE`.
- The Catalog service now stores only elements IDs and type (and their order in the course).
- An element should have a `visibility` field (either `PUBLIC`, `PROTECTED` or `PRIVATE`).
- A `page` entity can be created with all associated routes (`POST`, `PUT/{id}`, `GET/{id}`, `DELETE/{id}`).

### CODE REVIEW

**Comment by jlabatut** [comment]
User stories :

As | I want to be able to
-- | --
a contributor | create a course containing elements (videos, quizzes, pages) inside sections
a contributor | update a course
a contributor | delete a course
a member / a guest | access a course, browse its content (sections/elements)
an admin | manage the course (update/delete it)

File: src/resources/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+              - PRIVATE
+              - PROTECTED
+            default: PUBLIC
+        - name: userId
+          in: query
+          schema:
+            type: string
+      responses:
+        "200":
+          description: ""
```
Comment (kalil-pelissier) [inline src/resources/openapi.yml:145]: Remove the description if not needed.

File: src/resources/openapi.yml [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+      tags:
+        - Course
+    post:
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CourseBase"
+      responses:
```
Comment (kalil-pelissier) [inline src/resources/openapi.yml:173]: Missing Error response like (401, 403, 5XX)

File: src/resources/openapi.yml [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+    get:
+      parameters:
+        - name: id
+          required: true
+          in: path
+          schema:
+            type: string
+      responses:
+        "200":
+          description: ""
```
Comment (kalil-pelissier) [inline src/resources/openapi.yml:192]: Same remove the description if not needed.

File: src/resources/openapi.yml [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+          in: path
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/CourseBase"
+      responses:
```
Comment (kalil-pelissier) [inline src/resources/openapi.yml:212]: Missing Error response like (401, 403, 5XX)

File: src/resources/openapi.yml [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+    delete:
+      parameters:
+        - name: id
+          required: true
+          in: path
+          schema:
+            type: string
+      responses:
+        "204":
+          description: ""
```
Comment (kalil-pelissier) [inline src/resources/openapi.yml:298]: Remove the description if not needed.

**Review by kalil-pelissier — CHANGES_REQUESTED** [review]

**Review by kalil-pelissier — APPROVED** [review]

**Review by LeoFVO — APPROVED** [review]

**Pull request merged**